src/ProjectOrigin.PedersenCommitment/Ristretto/Oracle.cs

```diff
@@ -0,0 +1,48 @@
+namespace ProjectOrigin.PedersenCommitment.Ristretto;
+using System.Text;
+
+class Oracle
+{
+
+    private List<byte[]> messages = new List<byte[]>();
+
+    public Oracle(byte[] label)
+    {
+        messages.Add(label);
+    }
+
+    public void Domain(String domain)
+    {
+        var bytes = Encoding.UTF8.GetBytes("test");
+        messages.Add(bytes);
+    }
+
+
+    public void Add(params Point[] points)
```

Comment:
I would prefer a function that only allow to add single messages with an explicit label.
For example
```

public AddMessage(String label, byte[] message)
{
var byte_label = Encoding.UTF8.GetBytes(label);
messages.add(byte_label)
messages.add(message)
}

public AddPoint(String label, Point p)
{
var msg = p.Compress()._bytes
this.AddMessage(label, msg)
}
```
Note sure about the exact syntax (treat the above as pseudo code)